File: backend/Models/Recipe.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Threading.Tasks;

namespace backend.Models
{
    public class Recipe
    {
        [Key]
        public int Id { get; set; }
        [Required]
        public String Name { get; set; }
        [Required]
        public String Description { get; set; }
        [Required]
        public String Steps { get; set; }
        [Required]
        public bool ifExternal { get; set; }

        public ICollection<Rating> ratings { get; set; }
        public User user { get; set; }
        public ICollection<Photo> photos { get; set; }
        public ICollection<Recipe_ingredient> recipe_Ingredients { get; set; }
        public ICollection<User_history> user_Histories { get; set; }
    }
}

File: backend/Models/Recipe.java
package Models;

import javax.persistence.Entity;
import javax.persistence.GeneratedValue;

@Entity
public class Recipe
{
    @Id
    @GeneratedValue
    private int Id;

    private String Name;

    private String Description;

    private String Steps;

    private boolean ifExternal;

    public int getId() {
        return Id;
    }

    public void setId(int id) {
        Id = id;
    }

    public String getName() {
        return Name;
    }

    public void setName(String name) {
        Name = name;
    }

    public String getDescription() {
        return Description;
    }

    public void setDescription(String description) {
        Description = description;
    }

    public String getSteps() {
        return Steps;
    }

    public void setSteps(String steps) {
        Steps = steps;
    }

    public boolean isIfExternal() {
        return ifExternal;
    }

    public void setIfExternal(boolean ifExternal) {
        this.ifExternal = ifExternal;
    }
}